pkg/scheduler/framework/interface.go

```diff
@@ -379,7 +379,7 @@ type EnqueueExtensions interface {
 	// otherwise it would lead to undefined behavior.
 	//
 	// Appropriate implementation of this function will make Pod's re-scheduling accurate and performant.
-	EventsToRegister() []ClusterEventWithHint
+	EventsToRegister(context.Context) ([]ClusterEventWithHint, error)
```

Comment:
1. document what'll happen in case it returns errors.
2. eliminate a part: `needs to be static (not depend on configuration parameters)`. 
Rather, it should be OK that a plugin changes a behaviour based on configuration etc. NOT OK is a plugin changing a result after the scheduler starts (based on the cluster's state etc) because the scheduler evaluates it only once. @kubernetes/sig-scheduling-approvers Let me know if I'm missing something.

```go
	// When it returns an error, the scheduler fails to start.
	// Note: the returned list needs to be determined during a runtime,
	// and the scheduler only evaluates it once during start up.
	//
	// Appropriate implementation of this function will make Pod's re-scheduling accurate and performant.
	EventsToRegister(context.Context) ([]ClusterEventWithHint, error)
```